File: src/main/java/com/example/result/Ok.java
package com.example.result;

import com.example.option.None;
import com.example.option.Option;
import com.example.option.Some;

/** Successful result value. */
public final class Ok<T> implements Result<T> {
    private final T value;

    Ok(T value) {
        this.value = value;
    }

    @Override
    public boolean isOk() {
        return true;
    }

    @Override
    public Option<T> value() {
        return new Some<>(value);
    }

    @Override
    public Option<String> error() {
        return new None<>();
    }
}

File: src/main/node/magma/result/Ok.ts
import None from "../option/None";
import Option from "../option/Option";
import Some from "../option/Some";
/** Successful result value. */
export default class Ok<T> implements Result<T> {
    private readonly value: T;

    Ok(value: T): public {
        // TODO
    }

    @Override
    isOk(): boolean {
        return /* TODO */;
    }

    @Override
    value(): Option<T> {
        return new Some<T>(value);
    }

    @Override
    error(): Option<string> {
        return new None<string>();
    }
}
